inspace-nxtelem build errors in sim

When building for sim:nsh, there is a compilation error.

So when you `make menuconfig` and enable `INSPACE_TELEM` (including the dependencies first, `SENSORS`, `USENSOR`, and `UORB`), when running `make -j` you get the following compilation error:
```
CC:  src/pwm/pwm_olg.c src/syslogging.c: In function ‘setup_syslogging’:
CC:  pwd/lib_getpwbuf.c src/syslogging.c:18:31: error: ‘CONFIG_INSPACE_SYSLOG_PATH’ undeclared (first use in this function)
   18 |     __syslogging_file = fopen(CONFIG_INSPACE_SYSLOG_PATH, "a");
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/syslogging.c:18:31: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [/home/cmdr-johnalex/dev/university/CarletonURocketry/apps/Application.mk:239: src/syslogging.c.home.cmdr-johnalex.dev.university.CarletonURocketry.inspace-nxtelem.telemetry.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

--- a/telemetry/src/syslogging.c
+++ b/telemetry/src/syslogging.c
@@ -10,15 +10,17 @@ FILE *__syslogging_file;
 atomic_int syslog_count;
 
 /*
- * Sets up the syslogging to a file
- * @return Zero on success, or errno on failure
+ * Sets up the syslogging to a file if syslog output is enabled.
+ * @return Zero on success or skip, or errno on failure
  */
 int setup_syslogging(void) {
+#ifdef CONFIG_INSPACE_SYSLOG_PATH
     atomic_set(&syslog_count, 0);
     __syslogging_file = fopen(CONFIG_INSPACE_SYSLOG_PATH, "a");
     if (__syslogging_file == NULL) {
         return errno;
     }
+#endif
     return 0;
 }
 

